exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -0,0 +1,660 @@
+	data.Properties = make(map[string]string)
+	data.Measurements = make(map[string]float64)
+	data.ResponseCode, data.Success = getDefaultFormattedSpanStatus(span.Status())
+
+	switch incomingSpanType {
+	case httpSpanType:
+		fillRequestDataHTTP(span, data)
+	case rpcSpanType:
+		fillRequestDataRPC(span, data)
+	case messagingSpanType:
+		fillRequestDataMessaging(span, data)
+	case faasSpanType:
+		// TODO Implement the FaaS mapping
+	}
+
+	return data
+}
+
+// Maps Span to AppInsights RemoteDependencyData
+func spanToRemoteDependencyData(span pdata.Span, incomingSpanType spanType) *contracts.RemoteDependencyData {
+	// https://github.com/microsoft/ApplicationInsights-Go/blob/master/appinsights/contracts/remotedependencydata.go
+	// Start with some reasonable default for dependent spans.
+	data := contracts.NewRemoteDependencyData()
+	data.Id = idToHex(span.SpanID())
+	data.Name = span.Name()
+	data.ResultCode, data.Success = getDefaultFormattedSpanStatus(span.Status())
+	data.Duration = formatSpanDuration(span)
+	data.Properties = make(map[string]string)
+	data.Measurements = make(map[string]float64)
+
+	switch incomingSpanType {
+	case httpSpanType:
+		fillRemoteDependencyDataHTTP(span, data)
+	case rpcSpanType:
+		fillRemoteDependencyDataRPC(span, data)
+	case databaseSpanType:
+		fillRemoteDependencyDataDatabase(span, data)
+	case messagingSpanType:
+		fillRemoteDependencyDataMessaging(span, data)
+	case faasSpanType:
+		// TODO Implement the FaaS mapping
+	}
+
+	return data
+}
+
+// Maps Span to AppInsights MessageData
+func spanToMessageData(span pdata.Span, incomingSpanType spanType) *contracts.MessageData {
+	// https://github.com/microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/MessageData.bond
+	data := contracts.NewMessageData()
+	data.Properties = make(map[string]string)
+	data.Message = span.Name()
+
+	// TODO Is there a new version of the AppInsights contracts?
+	// The Measurements field is missing.
+	// For now, just copy all the string attributes into the message.properties.
+	if span.Attributes().Len() > 0 {
+		span.Attributes().ForEach(
+			func(k string, v pdata.AttributeValue) {
+				switch v.Type() {
+				case pdata.AttributeValueSTRING:
+					data.Properties[k] = v.StringVal()
+				}
+			})
+	}
+
+	return data
+}
+
+func getFormattedHTTPStatusValues(statusCode int64) (statusAsString string, success bool) {
+	if statusCode != 0 {
+		return strconv.FormatInt(statusCode, 10), statusCode <= 399
+	}
+
+	return "200", true
+}
+
+// Maps HTTP Server Span to AppInsights RequestData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#semantic-conventions-for-http-spans
+func fillRequestDataHTTP(span pdata.Span, data *contracts.RequestData) {
+	attrs := copyAndExtractHTTPAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	data.ResponseCode, data.Success = getFormattedHTTPStatusValues(attrs.HTTPStatusCode)
```

Comment:
What if user set only span.status without http_status_code attribute? then it will always be 200 success. can we take this attribute only if this attr exists?

Reply:
sure. I just won't touch data.ResponseCode or data.Success if the attribute isn't set